kingpin/actors/aws/ecs.py

```diff
@@ -181,19 +183,60 @@
 class ECSBaseActor(base.AWSBaseActor):
     """Base class for ECS actors."""
 
-    def _handle_failures(self, failures):
+    FAILURE_MISSING = 'MISSING'
+
+    def _handle_failures(self, failures, *ignorable):
         """Logs ECS failures. Raises a recoverable exception if there are any.
 
         Args:
             failures: list of failures from ECS api.
+            ignorable: list of failures to ignore.
         """
         if not failures:
             return
 
-        # TODO: When can we recover? What are the failures? How do we recover?
+        failure_reasons = []
+
         for failure in failures:
+            reason = failure['reason']
+            if reason in ignorable:
+                continue
+            failure_reasons.append(reason)
             self.log.error(failure)
-        raise exceptions.RecoverableActorFailure()
+        if failure_reasons:
+            raise exceptions.RecoverableActorFailure(failure_reasons)
+
+    @gen.coroutine
+    def _register_task(self, task_definition):
+        """Registers a task on ECS.
+
+        Args:
+            task_definition: dict of ECS task definition parameters.
+
+        Returns:
+            tuple: ('family', 'revision').
+        """
+
+        family = task_definition['family']
+        if self._dry:
+            self.log.info(
+                'Would register task definition with family {0}'.format(
+                    family))
+            raise gen.Return((family, 1))
+
+        self.log.info('Registering task definition with family {0}'.format(
+            family))
+        response = yield self.thread(
```

Comment:
What happens if this fails -- the inputs are bad, a 500 is raised, etc? I don't see any catching here or in the calling-code.